### ISSUE TITLE
RCS1234 false positive

### ISSUE THREAD

**Issue by josefpihrt** [dotnet/roslynator#1123]
**Product and Version Used**: 4.3.0

**Steps to Reproduce**:

```csharp
public enum Foo
{
    [Obsolete]
    Bar = 1,

    Baz = 1, // RCS1234
}
```

If `Bar` is obsolete it should not be used as a replacement for 1 because that would produce compiler warning.
